spectrum-network/src/protocol_handler/sync.rs

```diff
@@ -75,7 +77,29 @@ where
         self.outbox.push_back(SyncBehaviourOut::Send {
             peer_id,
             message: SyncMessage::SyncMessageV1(SyncMessageV1::GetPeers),
-        })
+        });
+
+        #[cfg(feature = "test_peer_punish_too_slow")]
+        {
+            // We send through extra messages to saturate the buffer to trigger peer-punishment for
+            // being `TooSlow`.
+            self.outbox.push_back(SyncBehaviourOut::Send {
+                peer_id,
+                message: SyncMessage::SyncMessageV1(SyncMessageV1::GetPeers),
+            });
+            self.outbox.push_back(SyncBehaviourOut::Send {
+                peer_id,
+                message: SyncMessage::SyncMessageV1(SyncMessageV1::GetPeers),
+            });
+            self.outbox.push_back(SyncBehaviourOut::Send {
+                peer_id,
+                message: SyncMessage::SyncMessageV1(SyncMessageV1::GetPeers),
+            });
+            self.outbox.push_back(SyncBehaviourOut::Send {
+                peer_id,
+                message: SyncMessage::SyncMessageV1(SyncMessageV1::GetPeers),
+            });
+        }
```

Comment:
Not sure mixing test code into prod code is a good idea, it will likely become quite messy and hard to navigate as codebase saturates. I'd propose to create a separate test impl of SyncBehaviour (or make test part pluggable into existing SyncBehaviour from the outside) with all the necessary logic and wire it into the node when assembling a test instance of it.

Reply:
I see you added FakeSyncBehaviour below, so why keeping this test part in the prod impl?

Reply:
Good idea, I've moved it to `FakeSyncBehaviour`.